From 47810c0cf2d0e6c3c7b2591e617463ef8bfb36d2 Mon Sep 17 00:00:00 2001
From: Marius Brehler <marius.brehler@amd.com>
Date: Mon, 29 Sep 2025 19:12:22 +0000
Subject: [PATCH] Fix finding libamd_smi.so

Allow amdsmi to find it's shared library bundled in Python packages
produced with TheRock.
---
 py-interface/amdsmi_wrapper.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/py-interface/amdsmi_wrapper.py b/py-interface/amdsmi_wrapper.py
index f5f05dfe..3ba2eb8b 100644
--- a/py-interface/amdsmi_wrapper.py
+++ b/py-interface/amdsmi_wrapper.py
@@ -167,6 +167,10 @@ _libraries = {}
 from pathlib import Path
 # libamd_smi.so can be located in several different places.
 # Look for it with below priority:
+# 0. Relative to amdsmi_wrapper.py in TheRock:
+#    `amdsmi_wrapper.py` is located in
+#    `_rocm_sdk_core/share/amd_smi/amdsmi`, libraries are in
+#    `_rocm_sdk_core/lib`.
 # 1. ROCM_HOME/ROCM_PATH environment variables
 #    - ROCM_HOME/lib
 #    - ROCM_PATH/lib (usually set to /opt/rocm/)
@@ -179,6 +183,9 @@ from pathlib import Path
 def find_smi_library():
     err = OSError("Could not load libamd_smi.so")
     possible_locations = []
+    # 0.
+    libamd_smi_path = Path(__file__).resolve().parent.parent.parent.parent / "lib/libamd_smi.so.26"
+    possible_locations.append(libamd_smi_path)
     # 1.
     rocm_path = os.getenv("ROCM_HOME", os.getenv("ROCM_PATH"))
     if rocm_path:
-- 
2.43.0

